runtime/src/accounts_db.rs

```diff
@@ -803,6 +882,14 @@ impl AccountsDB {
                                     );
                                 }
 
+                                let has_zero_lamport_accounts = locked_entry
+                                    .slot_list()
+                                    .iter()
+                                    .any(|(_slot, account_info)| account_info.lamports == 0);
+                                if !has_zero_lamport_accounts {
+                                    self.accounts_index.remove_zero_lamport(pubkey);
```

Comment:
nits: how about to name this `fn` to read better in this actual callsite like `self.accounts_index.forget_as_zero_lamport()` or `self.accounts_index.unmark_as_zero_lamport()`?